packages/core/plugin/src/withSentryAndroidGradlePlugin.ts

```diff
@@ -0,0 +1,96 @@
+  const autoUploadNativeSymbols = options.autoUploadNativeSymbols ?? true;
+  const includeNativeSources = options.includeNativeSources ?? true;
+  const includeSourceContext = options.includeSourceContext ?? false;
+  const autoInstallationEnabled = options.autoInstallationEnabled ?? false;
+
+  // Modify android/build.gradle
+  const withSentryProjectBuildGradle = (config: any): any => {
+    return withProjectBuildGradle(config, (projectBuildGradle: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (!projectBuildGradle.modResults || !projectBuildGradle.modResults.contents) {
+        throw new Error('android/build.gradle content is missing or undefined.');
+      }
+
+      const dependency = `classpath("io.sentry:sentry-android-gradle-plugin:${version}")`;
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (!projectBuildGradle.modResults.contents.includes(dependency)) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        projectBuildGradle.modResults.contents = projectBuildGradle.modResults.contents.replace(
+          /dependencies\s*{/,
+          `dependencies {\n        ${dependency}`,
+        );
+      }
+
+      return projectBuildGradle;
+    });
+  };
+
+  // Modify android/app/build.gradle
+  const withSentryAppBuildGradle = (config: any): any => {
+    return withAppBuildGradle(config, (config: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (config.modResults.language === 'groovy') {
+        const sentryPlugin = `apply plugin: "io.sentry.android.gradle"`;
+        const sentryConfig = `
+  sentry {
+      autoUploadProguardMapping = ${autoUploadProguardMapping}
+      includeProguardMapping = ${includeProguardMapping}
+      dexguardEnabled = ${dexguardEnabled}
+      uploadNativeSymbols = ${uploadNativeSymbols}
+      autoUploadNativeSymbols = ${autoUploadNativeSymbols}
+      includeNativeSources = ${includeNativeSources}
+      includeSourceContext = ${includeSourceContext}
+      autoInstallation {
+          enabled = ${autoInstallationEnabled}
+      }
+  }`;
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        let contents = config.modResults.contents;
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!contents.includes(sentryPlugin)) {
+          contents = `${sentryPlugin}\n${contents}`;
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!contents.includes('sentry {')) {
+          contents = `${contents}\n${sentryConfig}`;
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        config.modResults.contents = contents;
+      } else {
+        throw new Error('Cannot configure Sentry in android/app/build.gradle because it is not in Groovy.');
```

Comment:
Let's use the `warnOnce` helper instead of throw. It will make the message more readable for the users.

Reply:
Good idea 👍 
Updated with b1230bd8fb22efe3f2e6155fb246f42c2dbd20d7